barretenberg/cpp/pil/avm/gadgets/range_check.pil

```diff
@@ -0,0 +1,188 @@
+include "../main.pil";
+include "../fixed/powers.pil";
+
+namespace range_check(256);
+    pol commit clk;
+
+    // Range check selector
+    pol commit sel_rng_chk;
+    sel_rng_chk * (1 - sel_rng_chk) = 0;
+
+    // Witnesses 
+    // Value to range check
+    pol commit value;
+    // Number of bits to check against
+    pol commit rng_chk_bits;
+
+    // Bit Size Columns
+    // It is enforced (further down) that the selected column is the lowest multiple of 16 that is greater than rng_chk_bits 
```

Comment:
" ... the lowest multiple of 16 that is greater than rng_chk_bits " I think you meant "greater or equal".
A 128-bit range check is allowed and in this case is_lte_u128 would be 1.